### PR TITLE
task: Ban changes to variants through feature

### PR DESCRIPTION
**Pull request by chriswk**
After adding the new `/variants` endpoint for features we now have a way
to access control adding/modifying variants, so the /:featureName
endpoint should no longer allow editing/adding variants.

This removes variants as a known field from the featureMetadata schema
and tells joi to stripUnknown, thus making sure we never include
variants in the initial creation or future update calls.

For the old features v1 API we allow it to declare that it has already
validated the data coming with its own schema, so we should use the data
we get from it. Thus keeping the old v1 functionality intact

Co-authored-by: Simon Hornby <simon@getunleash.ai>

### CODE REVIEW

File: src/lib/routes/util.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ export const handleErrors: (
             return res.status(403).json(error).end();
         case 'MinimumOneEnvironmentError':
             return res.status(400).json(error).end();
+        case 'OperationDeniedError':
+            return res.status(403).json(error).end();
```
Comment (ivarconr) [inline src/lib/routes/util.ts:60]: We could group all the 403 cases together.

Comment (chriswk) [reply src/lib/routes/util.ts:60]: Agreed. I'll fix

Comment (chriswk) [reply src/lib/routes/util.ts:54]: I grouped statuses together in ascending order, so all 400, then 401, then 403, then 409.

Comment (ivarconr) [reply src/lib/routes/util.ts:54]: makes sense

**Review by ivarconr — APPROVED** [review]

**Pull request merged**